gifc/gifc.py

```diff
@@ -0,0 +1,478 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+# Interact with GitHub Gists using HTTP API
+
+# python3 -m gifc -h
+# ./gifc -h
+
+
+import os
+import subprocess
+import tempfile
+import yaml
+
+import gifc.arrowshiv as arrowshiv
+arrow = arrowshiv.importArrow()
+
+
+##############################################################################
+##############################################################################
+##############################################################################
+# gifc
+##############################################################################
+
+DEBUG = False
+ERROR_SUFFIX = "Please try using the Web GUI."
+TERM = os.environ.get('TERM', '')
+
+ENABLE_COLOR = False
+if '256color' in TERM.split('-'):
+    ENABLE_COLOR = True
+
+##############################################################################
+#  Config / Environment
+##############################################################################
+
+try:
+    EDITOR = os.environ.get('EDITOR', None)  # Default editor
+    FALLBACK_EDITOR = 'nano'  # ;_; wish it was vim <3
+    GH_USER = os.environ.get('GH_USER', None)
+    GH_TOKEN = os.environ.get('GH_TOKEN', None)
+    GH_BASE_URL = "https://api.github.com"
+    HOME_DIR = os.path.expanduser("~")
+    BINARY_DIR = os.path.dirname(os.path.realpath(__file__))
+    WORKING_DIR = os.getcwd(),
+except Exception as error:
+    emsg = "FATAL Error during initialization: %s" % (error)
+    print(emsg)
+    exit(emsg)
+
+# These are shared with __main__.py
+args = []
+argvars = {}
+session = {}
+
+##############################################################################
+# Helpers - Output
+##############################################################################
+
+
+def term_width():
+    ''' Determine Terminal width for output purposes '''
+    width = 80
+    try:
+        width = os.get_terminal_size()[0]
+    except Exception as e:
+        msg = "FATAL Error with console/terminal: %s" % (e)
+        printTerminal(msg)
+
+    msg = f"Terminal is {width} wide."  # noqa fstring
+    printDebug(msg)
+
+    return width
+
+
+class TMods:
+    ''' Terminal Output Modifiers '''
+
+    def __init__(self):
+        self.modifiers = {
+            '': '',
```

Comment:
For none, or null strings.  It's to prevent errors when interacting with this under certain logic paths.  Please let me know if I should go into further depth, but we want to make sure if someone selects 'no modifier', no modifier is used.